common4j/src/main/com/microsoft/identity/common/java/crypto/CryptoSuite.java

```diff
@@ -58,4 +61,14 @@
      * @return the signing algorithm desired by this suite.
      */
     SigningAlgorithm signingAlgorithm();
+
+    /**
+     * @return an AlgorithmParameterSpec if needed, null if not required.
+     */
+    AlgorithmParameterSpec cryptoSpec(Object... args);
```

Comment:
Per our conversation, this will be changed to Map, right? 
(If you couldn't get that in in time, please add javadoc on the expected arguments for each implementation / methods that takes Object ... as params - that would really help).

Reply:
I've put a lot of documentation in.